src/main/java/com/linecorp/armeria/server/docs/ThriftDocString.java

```diff
@@ -0,0 +1,170 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+import org.reflections.util.FilterBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * {@link ThriftDocString} is a DocString extractor for Thrift IDL JSON.
+ *
+ * To include docstrings in {@link DocService} pages, use a recent development version of the Thrift compiler
+ * (0.9.3 will not work) and compile your thrift files with 'json' code generation and
+ * include the resulting json files in the classpath location META-INF/armeria/thrift.
+ * The classpath location can be changed by setting the armeria.thrift.json.dir system property.
+ */
+final class ThriftDocString {
+
+    private static final Logger logger = LoggerFactory.getLogger(ThriftDocString.class);
+
+    private static final String THRIFT_JSON_PATH = getThriftJsonDirectory();
+
+    private static final TypeReference JSON_VALUE_TYPE = new TypeReference<HashMap<String, Object>>() {};
+
+    private static final String FQCN_DELIM = ".";
+
+    private static final String DELIM = "#";
+
+    private static Map<ClassLoader, Map<String, String>> cached = new ConcurrentHashMap<>();
+
+    private ThriftDocString() {
+    }
+
+    /**
+     * Extracts all DocStrings from all Thrift IDL JSON Resources.
+     * @return a map with key is FQCN and value is document string.
+     */
+    static Map<String, String> getAllDocStrings(ClassLoader classLoader) {
+        return cached.computeIfAbsent(classLoader, loader -> parseDocStrings(loader, getAllThriftJsons(loader)));
+    }
+
+    /**
+     * Parses DocStrings from input Thrift IDL JSON Resources.
+     * @return a map with key is FQCN and value is document string.
+     */
+    static Map<String, String> parseDocStrings(ClassLoader classLoader, Iterable<String> jsonPaths) {
+        final ImmutableMap.Builder<String, String> docStrings = ImmutableMap.builder();
+        for (String jsonPath : jsonPaths) {
+            docStrings.putAll(getDocStringsFromJsonResource(classLoader, jsonPath));
+        }
+        return docStrings.build();
+    }
+
+    // This is one-time codes for THRIFT_JSON_PATH. Do not call this later.
```

Comment:
Hmm - I disagree because 1) it's private and 2) it's not destructive or expensive even if for some reason it was called twice. If it's really that important, I'd recommend moving the contents into a static {} block instead.

But anyways, if you do want to keep this comment, then
```java
// Helper method to set up THRIFT_JSON_PATH, it should only be called once.
```